clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -770,8 +770,10 @@ void abortBatches(final RuntimeException reason) {
 
     /**
      * Abort any batches which have not been drained
+     * @return if one or more batches were aborted
      */
-    void abortUndrainedBatches(RuntimeException reason) {
+    boolean abortUndrainedBatches(RuntimeException reason) {
```

Comment:
This function is called in line 432 of Sender as well, should the return value be evaluated there?

Reply:
We only needed it here because the we were trying to ensure we had a chance to send the Abort before going into poll. After thinking about it, it seemed a little simpler to move the `hasAbortableError` check to `maybeSendAndPollTransactionalRequest`. Then we no longer need this change.